pkg/labels/labels.go

```diff
@@ -592,16 +592,18 @@ func (l Labels) FindReserved() Labels {
 //  src, next := parseSource("foo:bar==value")
 // Println(src) // foo
 // Println(next) // bar==value
-func parseSource(str string) (src, next string) {
+// For Cilium format 'delim' must be passed in as ':'
+// For k8s format 'delim' must be passed in as '.'
+func parseSource(str string, delim byte) (src, next string) {
 	if str == "" {
 		return "", ""
 	}
 	if str[0] == '$' {
 		return LabelSourceReserved, str[1:]
 	}
-	i := strings.IndexByte(str, ':')
+	i := strings.IndexByte(str, delim)
 	if i < 0 {
-		if strings.HasPrefix(str, LabelSourceReservedKeyPrefix) {
+		if delim != '.' && strings.HasPrefix(str, LabelSourceReservedKeyPrefix) {
```

Comment:
why not `delim == ':'`?

Reply:
This change merely avoids calling `string.HasPrefix` when it is known that it can not match. This only applies when `delim == '.'`, as we just concluded that `str` has no `'.'`, and `'.'` is the last character of `LabelSourceReservedKeyPrefix`. This logic does not apply for `delim == ':'`.